File: src/dataset/Java/luhn_exercism-ThomasZumsteg/claude/ClaudeSonnet4_Luhn_v1.java
import java.util.Arrays;

public class Luhn {
    private long code;

    public Luhn(Long code) {
        this.code = code;
    }

    public Luhn(Integer code){
        this.code = code.longValue();
    }

    public int getCheckDigit() {
        return (int)(code % 10L);
    }

    public int[] getAddends() {
        // Calculate length without string conversion
        long temp = code;
        int length = (temp == 0) ? 1 : 0;
        while (temp > 0) {
            length++;
            temp /= 10;
        }
        
        int[] intCode = new int[length];
        temp = code;
        
        // Process digits from right to left without string conversion
        for (int i = length - 1; i >= 0; i--) {
            int digit = (int)(temp % 10);
            temp /= 10;
            
            if ((length - i) % 2 == 1) {
                intCode[i] = digit;
            } else {
                int doubled = digit * 2;
                intCode[i] = (doubled > 9) ? doubled - 9 : doubled;
            }
        }
        
        System.out.format("%d ", code);
        System.out.println(intCode);
        return intCode;
    }

    public int getCheckSum() {
        // Calculate sum directly without creating array
        long temp = code;
        int length = (temp == 0) ? 1 : 0;
        while (temp > 0) {
            length++;
            temp /= 10;
        }
        
        temp = code;
        int sum = 0;
        
        for (int i = length - 1; i >= 0; i--) {
            int digit = (int)(temp % 10);
            temp /= 10;
            
            if ((length - i) % 2 == 1) {
                sum += digit;
            } else {
                int doubled = digit * 2;
                sum += (doubled > 9) ? doubled - 9 : doubled;
            }
        }
        
        return sum;
    }

    public boolean isValid() {
        return getCheckSum() % 10 == 0;
    }

    public static long create(long code) {
        int checkSum = new Luhn(code * 10).getCheckSum() % 10;
        return (10L * code) + (checkSum == 0 ? 0L : (10L - checkSum));
    }
}

File: src/dataset/c/luhn_exercism-fiplox/claude/ClaudeSonnet4_luhn_v1.c
#include "luhn.h"

#include <ctype.h>
#include <string.h>

bool luhn(const char *num)
{
	int sum = 0, n = 0;
	int digit;
	bool double_next = false;
	
	for (const char *p = num + strlen(num) - 1; p >= num; --p) {
		if (*p == ' ')
			continue;
		if (*p < '0' || *p > '9')
			return false;
		
		digit = *p - '0';
		n++;
		
		if (double_next) {
			digit <<= 1;
			if (digit > 9)
				digit -= 9;
		}
		sum += digit;
		double_next = !double_next;
	}
	return n > 1 && sum % 10 == 0;
}